### DISCUSSION TITLE
Why does the next.js Script tag with the beforeInteractive strategy don't load thirdParty script?

### DISCUSSION THREAD

**Discussion by JimboBilbo** [vercel/next.js#37098] [Help] [answered]
I can't get the next/script tag with beforeInteractive to work with in the _document.js.
See my full question on stackoverflow: https://stackoverflow.com/questions/72332146/why-does-the-next-js-script-tag-with-the-beforeinteractive-strategy-dont-load-t/72332560#72332560 
One person suggested to me that there is an issue when using next/script Tag in _document but i don't understand why.
Is it a bug or am i doing something wrong?

**Comment by xd-hearst** [comment, marked as answer]
Have the exact same issue.

Even the example in the next js official document does not work.

The following code sample is directly from  https://nextjs.org/docs/basic-features/script#beforeinteractive


```
// In _document.js
import { Html, Head, Main, NextScript } from 'next/document'
import Script from 'next/script'

export default function Document() {
  return (
    <Html>
      <Head />
      <body>
        <Main />
        <NextScript />
        <Script
          src="https://cdnjs.cloudflare.com/ajax/libs/lodash.js/4.17.20/lodash.min.js"
          strategy="beforeInteractive"
        ></Script>
      </body>
    </Html>
  )
}


```

**Reply by icyJoseph** [reply]
~~Yeah, this does not work at all, I thought it was because of React 18 new rendering root, but nope.~~

~~I found this issue: https://github.com/vercel/next.js/pull/37084#issue-1243838863~~

~~I'd also say that reading OP's SO post, I think OP is not grasping a few concepts, and is overall doing something in a complicated way.~~

~~That being said, this requires some investigation.~~

## Update

This has been fixed in https://github.com/vercel/next.js/pull/37000 - hopefully it gets released soon. This is the correct resolution, there was a bug, and it has been fixed.

**Reply by xd-hearst** [reply]
thanks. yeah. saw that as well. 

Seems now `beforeInteractive` needs to go into the `head`, based on the tests in the same Pr, also the pending pr https://github.com/vercel/next.js/pull/37084

https://github.com/vercel/next.js/pull/37000/files#diff-c2772fcf9a64ff0873a83bfbc3e33e0a097e391e6f439a63369e10140074634dR19-R25

**Reply by icyJoseph** [reply]
Ah right, blink and you miss it. Tested it also in canary. And the test does fail is it is not placed in the Head as child.
